front-end/src/app/shared/components/inputs/additional-info-input/additional-info-input.component.ts

```diff
@@ -1,13 +1,30 @@
-import { Component, Input } from '@angular/core';
+import { Component, OnInit, Input } from '@angular/core';
 import { BaseInputComponent } from '../base-input.component';
 
 @Component({
   selector: 'app-additional-info-input',
   templateUrl: './additional-info-input.component.html',
   styleUrls: ['./additional-info-input.component.scss'],
 })
-export class AdditionalInfoInputComponent extends BaseInputComponent {
+export class AdditionalInfoInputComponent extends BaseInputComponent implements OnInit {
   @Input() descriptionIsSystemGenerated = false;
   @Input() contributionPurposeDescriptionLabel = '';
   @Input() purposeDescriptionLabelNotice?: string;
+  @Input() purposeFieldTitle?: string; //Specifies a specific title for the CPD field
+  @Input() schedule?: 'A' | 'B'; //Specifies a schedule so that a default CPD title can be used
+
+  ngOnInit(): void {
+    if (!this.purposeFieldTitle) {
+      this.purposeFieldTitle = this.getPurposeFieldTitle();
+    }
+  }
+
+  public getPurposeFieldTitle() {
+    //This is a switch because there will *eventually* be many more cases
+    switch (this.schedule) {
+      case 'A':
+        return 'Purpose of Receipt';
```

Comment:
I think this needs to be all-caps

Reply:
It's not listed as all-caps in the ticket, but it's well worth double-checking

Reply:
ah you did the right thing in following it to the tee!  good to ask too.

Reply:
With so many people out, this'll have to wait for Monday so that I can double check this, but that's no big loss.